### DISCUSSION TITLE
How to get the currently selected node of the tree in the presentation library, as well as its parent node and child node?

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#444] [General] [answered]
Hi, I'm using the tree in the presentation library, but I can't get the node selected by the current mouse, as well as its parent node and child node. Can you help me? thank you

**Comment by pmconne** [comment]
@grigasp?

**Comment by grigasp** [comment, marked as answer]
Not sure what a "tree in the presentation library" means, since presentation library only provides helpers (React hooks & HOCs, data providers) for the `ControlledTree` component in `@bentley/ui-components` package. I would need to get more details on what exactly you're using to be able to help.

Also, you might find [UI Trees samples](https://www.itwinjs.org/sample-showcase/?group=UI+Trees&sample=unified-selection-tree-sample&imodel=Metrostation+Sample) useful.

**Reply by shenshengyi** [reply]
> Not sure what a "tree in the presentation library" means, since presentation library only provides helpers (React hooks & HOCs, data providers) for the `ControlledTree` component in `@bentley/ui-components` package. I would need to get more details on what exactly you're using to be able to help.
> 
> Also, you might find [UI Trees samples](https://www.itwinjs.org/sample-showcase/?group=UI+Trees&sample=unified-selection-tree-sample&imodel=Metrostation+Sample) useful.



Hello, I am using controlledtree, which displays imodel data through ruleset.json Configured

Now I want to get the currently selected treeitemnode and get its parent node and child node, but I can't get it now.

**Reply by grigasp** [reply]
You can get to already loaded nodes using `TreeModel` - that should work for getting parent node. Children might not be loaded - to get them you need to use `TreeNodeLoader`. To handle tree node selection events you will need to supply your own custom `TreeEventHandler` implementation. There's an example for how to create one [here](https://www.itwinjs.org/sample-showcase/?group=UI+Trees&sample=custom-event-handler-tree-sample).

**Reply by shenshengyi** [reply]
OK, thank you very much. This will be the answer I need.
